src/utils/dashboardUtils.ts

```diff
@@ -0,0 +1,66 @@
+import { DashboardContentOptions } from "amazon-quicksight-embedding-sdk";
+import { Logger } from "./logger";
+
+/**
+ * Constructs and returns an array of QuickSight parameter objects for a user's studies.
+ *
+ * - If the user's first study is `All`, the function returns an empty array (allowing QuickSight to display all data).
+ * - If the user has a valid array of studies, it creates a `studiesParameter` whose values are the `_id` fields of each study.
+ * - Otherwise, it logs an error and returns a parameter array with `["NO-CONTENT"]`.
+ *
+ *
+ * @param {User} user - The current user
+ * @returns {DashboardContentOptions["parameters"]} The updated dashboard parameters
+ */
+export const addStudiesParameter = (user: User): DashboardContentOptions["parameters"] => {
+  const params: DashboardContentOptions["parameters"] = [];
+  const { studies } = user || {};
+
+  // If "All" is the first study, do NOT push the "studiesParameter"
+  if (studies?.[0]?._id === "All") {
```

Comment:
Technically this assertion is correct, the "All" study should be the only one assigned and in the first index. However, I suspect there may be instances where this isn't true.

e.g. If a Federal Lead has permission to create a Submission Request, if that request is approved, I suspect BE would assign the newly created study to the Federal Lead's profile. If the study happens to be inserted at the first spot, that would break this logic.

The chances of these sort of issues happening is slim, but if you agree that it could happen, maybe let's just search the entire array of studies?

I did something like this in the DS Creation dialog, feel free to adapt:

```javascript
(studies || [])?.findIndex((s) => s?._id === "All") !== -1
```